pkg/deploy/lattice/service_manager_test.go

```diff
@@ -3,1035 +3,296 @@ package lattice
-			wantErr:                          errors.New(LATTICE_RETRY),
-			wantListServiceOutput:            []*vpclattice.ServiceSummary{},
-		},
-		{
-			name:                             "test-5",
-			tags:                             nil,
-			meshName:                         "test-mesh-5",
-			meshId:                           "id-234567890",
-			meshArn:                          "arn-234567890",
-			wantServiceName:                  "svc-test-5",
-			wantServiceId:                    "id-123456789",
-			wantServiceArn:                   "arn-123456789",
-			wantMeshServiceAssociationStatus: vpclattice.ServiceNetworkServiceAssociationStatusDeleteFailed,
-			wantErr:                          errors.New(LATTICE_RETRY),
-			wantListServiceOutput:            []*vpclattice.ServiceSummary{},
-		},
-	}
-
-	for _, tt := range tests {
-		fmt.Printf("testing >>> %v \n", tt.name)
-		c := gomock.NewController(t)
-		defer c.Finish()
-		ctx := context.TODO()
-		mockVpcLatticeSess := mocks.NewMockLattice(c)
-		latticeDataStore := latticestore.NewLatticeDataStore()
-		latticeDataStore.AddServiceNetwork(tt.meshName, config.AccountID, tt.meshArn, tt.meshId, latticestore.DATASTORE_SERVICE_NETWORK_CREATED)
-		mockCloud := mocks_aws.NewMockCloud(c)
-
-		createServiceOutput := &vpclattice.CreateServiceOutput{
-			Arn:    &tt.wantServiceArn,
-			Id:     &tt.wantServiceId,
-			Name:   &tt.wantServiceName,
-			Status: aws.String(vpclattice.ServiceStatusActive),
-		}
-		createServiceNetworkServiceAssociationOutput := &vpclattice.CreateServiceNetworkServiceAssociationOutput{
-			Arn:      nil,
-			DnsEntry: nil,
-			Id:       nil,
-			Status:   &tt.wantMeshServiceAssociationStatus,
-		}
-		input := &latticemodel.Service{
+func TestServiceManagerInteg(t *testing.T) {
+	c := gomock.NewController(t)
+	defer c.Finish()
+
+	lat := services.NewMockLattice(c)
+	cfg := mocks_aws.CloudConfig{VpcId: "vpc-id", AccountId: "account-id"}
+	cl := mocks_aws.NewDefaultCloud(lat, nil, cfg)
+	ds := latticestore.NewLatticeDataStore()
+	ctx := context.Background()
+	m := NewServiceManager(cl, ds)
+
+	// Case for single service and single sn-svc association
+	// Make sure that we send requests to Lattice for create Service and create Sn-Svc
+	t.Run("create new service and association", func(t *testing.T) {
+		ds.AddServiceNetwork("sn", cfg.AccountId, "sn-arn", "sn-id", "sn-status")
+
+		// service does not exists in lattice
+		lat.EXPECT().
+			ListServicesAsList(gomock.Any(), gomock.Any()).
+			Return([]*SvcSummary{}, nil)
+
+		// assert that we call create service
+		lat.EXPECT().
+			CreateServiceWithContext(gomock.Any(), gomock.Any()).
+			Return(&CreateSvcResp{
+				Arn:      aws.String("arn"),
+				DnsEntry: &vpclattice.DnsEntry{DomainName: aws.String("dns")},
+				Id:       aws.String("svc-id"),
+			}, nil)
+
+		// assert that we call create association
+		lat.EXPECT().
+			CreateServiceNetworkServiceAssociationWithContext(gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
Would be nice to include some assertions in this test on arguments

Reply:
Request part is covered in smaller unit test TestCreateSvcReq